numbat/examples/inspect.rs

```diff
@@ -86,25 +79,137 @@ fn inspect_functions_in_module(ctx: &Context, module: String) {
         println!("```");
         println!();
+
+        if !examples.is_empty() {
+            println!("<details>");
+            println!("<summary>Examples</summary>");
+            println!();
+
+            for (example_code, example_description) in examples {
+                let mut example_ctx = prelude_ctx.clone();
+                let extra_import = if !example_ctx
+                    .resolver()
+                    .imported_modules
+                    .contains(&module_path)
+                {
+                    format!("use {}\n", module)
+                } else {
+                    "".into()
+                };
+                let _result = example_ctx
+                    .interpret(&extra_import, CodeSource::Internal)
+                    .unwrap();
+
+                if let Ok((statements, results)) =
+                    example_ctx.interpret(&example_code, CodeSource::Internal)
+                {
+                    //Format the example input
+                    let example_input = format!(">>> {}", example_code);
+
+                    //Encode the example url
+                    let url_code = extra_import + &example_code;
+                    let example_url = format!(
+                        "https://numbat.dev/?q={}",
+                        percent_encoding::utf8_percent_encode(
+                            &url_code,
+                            percent_encoding::NON_ALPHANUMERIC
+                        )
+                    );
+
+                    //Assemble the example output
+                    let mut example_output = String::new();
+                    example_output += "\n";
+
+                    for statement in &statements {
+                        example_output += &plain_text_format(&statement.pretty_print(), true);
+                        example_output += "\n\n";
+                    }
+
+                    let result_markup = results.to_markup(
+                        statements.last(),
+                        &example_ctx.dimension_registry(),
+                        true,
+                        true,
+                    );
+                    example_output += &plain_text_format(&result_markup, false);
+
+                    if results.is_value() {
+                        example_output += "\n";
+                    }
+
+                    //Print the example
+                    if let Some(example_description) = example_description {
+                        println!(
+                        "* {}\n\n  <a href=\"{}\"><i class=\"fa fa-play\"></i> Run this example</a>",
+                        replace_equation_delimiters(example_description),
+                        example_url
+                    );
+                    } else {
+                        println!(
+                            "* <a href=\"{}\"><i class=\"fa fa-play\"></i> Run this example</a>\n",
+                            example_url
+                        );
+                    }
```

Comment:
Can we please move the "Run this example" links *below* the code snippets? I think that would look better because the "play" icons wouldn't line up with the bullets. And most of the time, users will only read the example output instead of trying it interactively.

![image](https://github.com/user-attachments/assets/efdc5bf2-04cf-4d4e-94a9-3b3490a26753)

Reply:
Sure, that is fine for me.

Reply:
Actually, after trying it out I think it doesn't look exactly nice with the "Run this example" links below the code snippets either, at least when there is no example description. Maybe we should not use a bullet list here at all.

![image](https://github.com/user-attachments/assets/79052f52-f48e-417d-8573-4dea60e7b87e)

Reply:
I changed the display and tried to mimic the mdBook Rust Playground feature "play button":

![image](https://github.com/user-attachments/assets/78333c6a-0152-4959-9002-412ca153f948)

The implementation is a little hacky as it requires to insert html code blocks in the markdown documents, however since they are entirely autogenerated I think it should be okay.

Reply:
Really nice! Thank you :+1: